### ISSUE TITLE
Mod incompatibility with... minetogether?

### ISSUE THREAD

**Issue by zedekblue** [hypherionmc/sdlink#25]
Tried tossing this into my stoneblock 3 v1.3.0 server and had some very early crashes on startup.  
Turns out it has a problem with minetogether?  
I've always just removed/disabled the mod, I'm not really sure why its in the server files for this pack to begin with, but I thought maybe it could be something you'd be interested in resolving.  
Using sdlink Forge v 2.0.1

**Comment by zedekblue** [comment]
seems to have some more unresolved issues even with minetogether removed (albeit I'm not sure how graceful it was to just delete - I did seem to get an error saying it was missing during startup)

This time I had an actual error log to show for it. I'm not sure how worth it even is to share considering I am blindly asking for compatibility with 100+ something mods on a version that gets 0.1% of your traffic, but I'll leave that for you to decide.
```
[17Nov2022 09:29:11.938] [Server thread/ERROR] [net.minecraftforge.eventbus.EventBus/EVENTBUS]: Exception caught during firing event: Cannot read field "broadcastCommands" because "me.hypherionmc.sdlinklib.config.ConfigController.modConfig.chatConfig" is null
	Index: 3
	Listeners:
		0: HIGH
		1: ASM: class dev.architectury.event.forge.EventHandlerImplCommon event(Lnet/minecraftforge/event/CommandEvent;)V
		2: NORMAL
		3: ASM: me.hypherionmc.sdlink.ForgeEventHandler@53f816b commandEvent(Lnet/minecraftforge/event/CommandEvent;)V
		4: ASM: class de.srendi.advancedperipherals.common.events.Events onCommand(Lnet/minecraftforge/event/CommandEvent;)V
java.lang.NullPointerException: Cannot read field "broadcastCommands" because "me.hypherionmc.sdlinklib.config.ConfigController.modConfig.chatConfig" is null
	at TRANSFORMER/sdlink@2.0.1/me.hypherionmc.sdlink.server.ServerEvents.commandEvent(ServerEvents.java:145)
	at TRANSFORMER/sdlink@2.0.1/me.hypherionmc.sdlink.ForgeEventHandler.commandEvent(ForgeEventHandler.java:66)
	at net.minecraftforge.eventbus.ASMEventHandler_35_ForgeEventHandler_commandEvent_CommandEvent.invoke(.dynamic)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
	at MC-BOOTSTRAP/eventbus@5.0.3/net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.commands.Commands.m_82117_(Commands.java:226)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.server.dedicated.DedicatedServer.m_139665_(DedicatedServer.java:341)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:327)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:842)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:685)
	at TRANSFORMER/minecraft@1.18.2/net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:261)
	at java.base/java.lang.Thread.run(Unknown Source)
```

I also got this error repeated every time I tried to run the `stop` command, and had to kill the server at this point.

**Comment by hypherionmc** [comment]
Hey.

Yeah the Minetogether issue is something I sorta know about. I ran into this with one of Vazkii's modpacks. I also ended up just removing the mod, cause I didn't need it.

I did try fixing it on my end, but no matter what I do, Forge just kept complaining about the conflicting dependencies, even though both mods repack them correctly. I gave up after the 5th try and never tried again, cause this is honestly the first issue report I ever got for it, so I guess either no one else encountered this, or they just followed the same path as us.


As for your deleted comment, I think that might either be the config being corrupt (due to the startup crash), or there is an actual error where the config is unloaded, before the stop command finishes executing.

If that's a consistent error, please do let me know and pass along all the logs that you can.

**Comment by zedekblue** [comment]
yeah your forge downloads don't seem to get much traffic so I'm not surprised not many have reported/complained about it.  
Also yes, deleted comment was unrelated after all and just a config issue on my part (last config ver I used (7) didn't seem to need both chat and server webhook url filled in). Deleted to keep this on topic haha.  
  
I can't quite tell from the documentation if minetogether even needs to be installed server side. I haven't had any complaints yet so perhaps its a non issue to begin with.

**Comment by hypherionmc** [comment]
All good. That problem may be related to issue #24. It shouldn't be required to set both webhooks.

And from my own experience, minetogether works fine on the client if it's missing on the server, so it should be good

**Comment by hypherionmc** [comment]
So to update, I attempted once again to fix this compat issue, but I have the feeling they have to fix it from their side as well, cause the "conflicting module" is definitely repacked correctly on our side, so I'm kinda at a loss about what to do

**Comment by zedekblue** [comment]
https://github.com/CreeperHost/MineTogether/issues

**Comment by hypherionmc** [comment]
This issue should now finally be fixed since 2.2.3
